neqo-transport/src/crypto.rs

```diff
@@ -637,10 +637,11 @@ impl CryptoDxState {
             hex(body)
         );
         // The numbers in `Self::limit` assume a maximum packet size of 2^11.
-        if body.len() > 2048 {
-            debug_assert!(false);
-            return Err(Error::InternalError);
-        }
+        // FIXME: Figure out how to allow packets > 2048.
+        // if body.len() > 2048 {
+        //     debug_assert!(false);
+        //     return Err(Error::InternalError);
+        // }
```

Comment:
Ping @martinthomson

Reply:
@martinthomson what to do about this?

Reply:
Good question.  I missed this in my first pass.

My suggestion would be to change `Self::limit()` to `self.limit()` instead, with a variable on the object and a check here to update the value if necessary.

What we want is to halve the current limit for every bit that is set starting from bit 11 (i.e., 2048).  You can use `u64::leading_zeros` to compute that efficiently.  Once you halve the limit, you should not increase it again.

We'd want to assert (debug_assert, probably) that you don't exceed 2^16 bits of data.

An alternative approach is to shift to counting the total number of 16 byte blocks (rounded up!)  You would increase the limit by a factor of 2^7 as a result.